### ISSUE TITLE
Tests run when compile fails

### ISSUE THREAD

**Issue by BebeSparkelSparkel** [monounity/karma-typescript#215]
It seems that the tests shouldn't run if the compile fails.

Is there a way to block the test from running if the ts => js fails?

**Comment by erikbarke** [comment]
Check out `noEmitOnError` in the readme, if you set it to `true` in, for instance `karmaTypescriptConfig.compilerOptions` the Karma process will exit if there are compilation errors!

**Comment by BebeSparkelSparkel** [comment]
That is better, but it kills my karma process when I have `singleRun: false,`.

[karma-tslint](https://www.npmjs.com/package/karma-tslint) has a good option `stopOnFailure: true` that stops the current test but doesn't kill the process

**Comment by erikbarke** [comment]
There's a fix for this in #216!

**Comment by BebeSparkelSparkel** [comment]
Thanks!

**Comment by fr0** [comment]
I'm seeing this issue (or a similar one) in `3.0.13`... any compiler error in my test .ts code causes the karma process to exit. Is this a regression?

```
06 11 2018 14:56:35.069:ERROR [karma-server]: { Error: COMPILATION ERROR
    at maybeWrapAsError (/Users/crfrolik/Documents/atlas-client-web/node_modules/bluebird/js/release/util.js:61:12)
    at /Users/crfrolik/Documents/atlas-client-web/node_modules/bluebird/js/release/nodeback.js:38:50
    at nextPreprocessor (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma/lib/preprocessor.js:22:14)
    at Object.callback (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/karma/preprocessor.js:19:36)
    at /Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:73:20
    at Array.forEach (<anonymous>)
    at Compiler.onProgramCompiled (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:66:24)
    at Compiler.compileProject (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:52:14)
    at Compiler.<anonymous> (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:28:23)
    at invokeFunc (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/node_modules/lodash/lodash.js:10325:23)
    at trailingEdge (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/node_modules/lodash/lodash.js:10374:18)
    at Timeout.timerExpired [as _onTimeout] (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/node_modules/lodash/lodash.js:10362:18)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)
  cause: Error: COMPILATION ERROR
    at maybeWrapAsError (/Users/crfrolik/Documents/atlas-client-web/node_modules/bluebird/js/release/util.js:61:12)
    at /Users/crfrolik/Documents/atlas-client-web/node_modules/bluebird/js/release/nodeback.js:38:50
    at nextPreprocessor (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma/lib/preprocessor.js:22:14)
    at Object.callback (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/karma/preprocessor.js:19:36)
    at /Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:73:20
    at Array.forEach (<anonymous>)
    at Compiler.onProgramCompiled (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:66:24)
    at Compiler.compileProject (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:52:14)
    at Compiler.<anonymous> (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/dist/compiler/compiler.js:28:23)
    at invokeFunc (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/node_modules/lodash/lodash.js:10325:23)
    at trailingEdge (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/node_modules/lodash/lodash.js:10374:18)
    at Timeout.timerExpired [as _onTimeout] (/Users/crfrolik/Documents/atlas-client-web/node_modules/karma-typescript/node_modules/lodash/lodash.js:10362:18)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5),
  isOperational: true }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by erikbarke** [comment]
It's behaving exactly as requested in this issue, added in #216. However, this seems like a controversial behavior since it's really hard to get Karma to just stop the test run without crashing, as reported in #274. #274 needs to be fixed before releasing 4.0.0, since the fix would (probably) be a another breaking change. I'm currently digging deep into #297, which also should be released 4.0.0, so any help with #274 would be awesome.
